File: 프로그래머스/unrated/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    dx = [-1,0,0,1]
    dy = [0,-1,1,0]
    for i in range(4):
        nx = w+dx[i]
        ny = h+dy[i]
        if 0<=nx<len(board[0]) and 0<=ny<len(board):
            if board[ny][nx] == board[h][w]:
                answer+=1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
import java.util.*;

class Solution {
    static int[] dx = {-1, 0, 0, 1};
    static int[] dy = {0, -1, 1, 0};
    
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        int n = board.length;
        String color = board[h][w];
        
        for (int i = 0; i < 4; i++) {
            int nx = h+dx[i];
            int ny = w+dy[i];
            if (nx>=0 && nx<n && ny>=0 && ny<n && board[nx][ny].equals(color)) answer++;
        }
        return answer;
    }
}